### DISCUSSION TITLE
Values of the network training diagnostic parameters

### DISCUSSION THREAD

**Discussion by setar** [AI4Finance-Foundation/FinRL#127] [General]
Hi
I am currently training a network based on env_multistock_cashpenalty.ipynb with cryptocurrency source data.
Please explain where to read the values of the debug log to understand the quality of training and what to strive for?

```
 457| 499|update         |$924,288       |$1,013,781     |0.00282%  |1.37813%  |91.17%    
 457| 999|update         |$848,272       |$1,013,387     |0.00136%  |1.33867%  |83.71%    
 457|1499|update         |$772,150       |$1,110,341     |0.00755%  |11.03412% |69.54%    
 457|1999|update         |$695,948       |$1,073,990     |0.00373%  |7.39902%  |64.80%    
 457|2499|update         |$619,725       |$1,291,825     |0.01151%  |29.18255% |47.97%    
 457|2999|update         |$543,497       |$1,686,729     |0.02290%  |68.67287% |32.22%    
 457|3256|Last Date      |$504,366       |$2,022,736     |0.03141%  |102.27357%|24.93%    
Eval num_timesteps=229000, episode_reward=0.28 +/- 0.00
Episode length: 3257.00 +/- 0.00
------------------------------------------
| environment/            |              |
|    GainLoss_pct         | 102          |
|    avg_daily_trades     | 2.31         |
|    avg_daily_trades_... | 0.771        |
|    cash_proportion      | 0.249        |
|    completed_steps      | 3256         |
|    sum_rewards          | 0.275        |
|    total_assets         | 2022735      |
|    total_reward_pct     | 102          |
|    total_trades         | 7.53e+03     |
| eval/                   |              |
|    mean_ep_length       | 3.26e+03     |
|    mean_reward          | 0.275        |
| time/                   |              |
|    fps                  | 24           |
|    iterations           | 895          |
|    time_elapsed         | 9222         |
|    total_timesteps      | 229120       |
| train/                  |              |
|    approx_kl            | 0.0019373121 |
|    clip_fraction        | 0            |
|    clip_range           | 0.2          |
|    entropy_loss         | -4.26        |
|    explained_variance   | 5.96e-08     |
|    learning_rate        | 5e-06        |
|    loss                 | -0.00143     |
|    n_updates            | 8940         |
|    policy_gradient_loss | -2.01e-05    |
|    std                  | 0.999        |
|    value_loss           | 0.000206     |
------------------------------------------
```

**Comment by george-adams1** [comment]
@Setar Pretty much everything under train/

I think looking at the loss would be particularly useful in this case. Just do a quick Google search for what a good value for each parameter is and then improve your model based on that.
